torch/testing/__init__.py

```diff
@@ -70,7 +70,8 @@ def _unravel_index(flat_index, shape):
 #
 #   Bool tensors are equal only if they are identical, regardless of
 #   the rtol and atol values.
-def _compare_tensors_internal(a: torch.Tensor, b: torch.Tensor, *, rtol, atol, equal_nan: bool) -> _compare_return_type:
+def _compare_tensors_internal(a: torch.Tensor, b: torch.Tensor, *, rtol, atol, equal_nan: Union[str, bool]) -> _compare_return_type:
```

Comment:
These equal_nan changes look reasonable but they seem like a separate PR

Reply:
The `equal_nan` is THE thing for this PR, and the splitting of test look like a separate PR :)

Reply:
haha, OK. However you'd like to split them.

Reply:
How about not splitting them? This PR fixes two failing tests, and the side effect is to split existing test suit, and the reason for having this side effect is because this provides better scope for the `"relaxed"`.

Reply:
OK; if we improve the test granularity we can combine them.